internal/ibazel/fswatcher/fsevents/fsevents.go

```diff
@@ -96,6 +99,34 @@ func newEvent(name string, mask fsevents.EventFlags) (common.Event, bool) {
 	return e, true
 }
 
+// Find the longest common root path of all directories to watch.
+func findCommonRoot(names []string) []string {
+	if len(names) == 0 {
+		return []string{}
+	}
+
+	rootSplit := strings.Split(strings.Trim(names[0], "/"), "/")
+	rootLength := len(rootSplit)
+
+	for _, dir := range names {
+		split := strings.Split(strings.Trim(dir, "/"), "/")
+		commonLength := 0
+		for i := 0; i < rootLength && i < len(split); i++ {
+			if rootSplit[i] != split[i] {
+				break
+			}
+			commonLength = i + 1
+		}
+		rootLength = commonLength
+	}
+
+	if rootLength == 0 {
+		return []string{"/"}
```

Comment:
Maybe this should throw an error? Or, maybe we should throw an error if the resolve to outside the workspace root?

Reply:
I'm generally in favor of functions being able to return errors. I think that's a good use case for an error right here

Reply:
done 👍